### DISCUSSION TITLE
Potential broken release: 1.3.24

### DISCUSSION THREAD

**Discussion by jakabk** [sqlalchemy/sqlalchemy#6282] [Usage Questions]
I cannot write a reproducible example, because I don't understand sqlalchemy inner logic, but I have many similar errors when importing related models.

```
.tox/py35/lib/python3.5/site-packages/sqlalchemy/sql/schema.py:1534: in __init__
    self.type._set_parent_with_dispatch(self)
.tox/py35/lib/python3.5/site-packages/sqlalchemy/sql/type_api.py:1007: in _set_parent_with_dispatch
    parent, outer=True
.tox/py35/lib/python3.5/site-packages/sqlalchemy/sql/base.py:461: in _set_parent_with_dispatch
    self._set_parent(parent, **kw)
E   TypeError: _set_parent() got an unexpected keyword argument 'outer'
```
I think the problem is in this row:
https://github.com/sqlalchemy/sqlalchemy/blob/ee17cbad9ed4adbc1a6a1b6273ad1ded66e25d99/lib/sqlalchemy/sql/base.py#L456
Missing ```**kw```

At least I would accept how to write a reproducible example (what should I write to trigger calling this inner functions: _set_parent_with_dispatch).

**Comment by zzzeek** [comment]
this is likely regression in 1.4 as well

**Reply by zzzeek** [reply]
oh, no seems to be in 1.3 only.   let's get a test case

**Reply by zzzeek** [reply]
do you have a custom SchemaType in use ?

**Reply by zzzeek** [reply]
that's a really weird failure, because "self" there is a TypeDecorator and it overrides ._set_parent().  I would need to see an example of the kinds of datatypes you're using.  I can't get it to reach that codepath.

**Reply by jakabk** [reply]
I have the issue with 1.4.7 as well.
Yes, we use ["sqlalchemy_enum34"](https://github.com/spoqa/sqlalchemy-enum34/blob/4aebcc339d2b85e7a50ae6996c7453ca993db6d7/sqlalchemy_enum34.py#L16) package which uses SchemaType.

**Reply by zzzeek** [reply]
but 1.4 is not missing the **kw from set_parent() AFAICT

**Reply by zzzeek** [reply]
please supply an example use

**Reply by zzzeek** [reply]
oh, it looks like this perhaps?  

https://github.com/spoqa/sqlalchemy-enum34/blob/4aebcc339d2b85e7a50ae6996c7453ca993db6d7/sqlalchemy_enum34.py#L56

maybe you can try reporting a bug at https://github.com/spoqa/sqlalchemy-enum34/issues

**Reply by jakabk** [reply]
Ok, you are right it is a sqlalchemyy_enum34 issue, I've dropped it and used sqlalchemy Enum type and this solved my issue.
Thank you for the help, I was totally lost.

Perhaps in 1.3.24 there is an issue with the missing `**kw`.
In 1.4.x it's ok.

**Reply by zzzeek** [reply]
yes it's still "wrong" in 1.3, but there's no codepath I can find that goes there, so i will fix it, but it wont fix your problem

**Reply by zzzeek** [reply]
the sqlalchemy_enum34 package fixed this here:   https://github.com/spoqa/sqlalchemy-enum34/commit/aafc37a4390a5048e33df7f0e5a3f38b0abfdd9e   this is actually the fix.  download version 2.0.0 and it works

**Reply by jakabk** [reply]
> yes it's still "wrong" in 1.3, but there's no codepath I can find that goes there, so i will fix it, but it wont fix your problem

Yes, I understand.

Thank you very much it was a lot of help. My issue is solved in many ways. (dropping sqlalchemy-enum34 and using sqlalchemy enum type or use the new version of sqlalchemy-enum34)
